src/frontend/src/pages/Auth/Callback.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { useNavigate, useSearchParams } from "react-router";
+
+import { setToken } from "reducers/auth/authSlice";
+import { getApiToken } from "api/auth";
+import { LOCAL_STORAGE_TOKEN_KEY } from "src/constants";
+import useSupportedLanguage from "hooks/useSupportedLanguage";
+
+const LoginCallback = () => {
+    const [searchParams] = useSearchParams();
+    const navigate = useNavigate();
+    const dispatch = useDispatch();
+    const language = useSupportedLanguage();
+
+    useEffect(() => {
+        // This function will run once when the component mounts
+        const processToken = async () => {
+            // Grab the auth token from the URL
+            const authToken = searchParams.get("echo_auth");
+
+            // If no token is found, go back to sign in page.
+            if (!authToken) {
+                navigate(`/${language}/login`, { replace: true });
+                return;
+            }
+
+            try {
+                // Call the backend API to exchange auth token with API token
+                const response = await getApiToken(authToken);
+                const { token: apiToken } = response;
+                // Add the API token to localStorage for persistence
+                localStorage.setItem(LOCAL_STORAGE_TOKEN_KEY, apiToken);
+                // Add the API token to the Redux state
+                dispatch(setToken(apiToken));
+                // Go to the /discover page
+                navigate(`/${language}/discover`, { replace: true });
```

Comment:
Right now, it goes to the discover page. I think user needs to set up a profile so that profile-related stuffs can show up in the discover page. I suggest we handle that on the discover page, because discover page is where user profile settings are needed, plus it covers both cases when user is logged in or not
- Grab user profiles from API
- If no profile at all or if the preferences are not complete, we take them to the user profiles page to go through the wizard
- Otherwise, biz as usual.

What do you think?

Reply:
Yeah that sounds like the right plan 👍 as part of that sign-in flow we'll need to fetch all the other data too so in detail that looks like:
- On successful sign-in: fetch neighborhoods, neighborhood bounds, networks data and user profile

-  If a user profile exists, fetch paths and times data for all destinations and set active (primary) destination in state
       -   Once paths and times data is ready, dispatch calculate ranking logic
-  If a user profile does not exist (or destinations list is empty), take them to the user profiles page
        -   On save, fetch paths and times data and calculate ranking